### ISSUE TITLE
Incompatible record argument used in a function call

### ISSUE THREAD

**Issue by Daiwen** [modelica/ModelicaStandardLibrary#3483]
In `Modelica.Fluid.Dissipation.PressureLoss.StraightPipe.dp_twoPhaseOverall_DP` we define `DP_fric` as follow:

      SI.Pressure DP_fric = if IN_con.frictionalPressureLoss == TYP.Friedel then Modelica.Fluid.Dissipation.Utilities.Functions.PressureLoss.TwoPhase.dp_twoPhaseFriedel_DP(IN_con, IN_var, m_flow) else if IN_con.frictionalPressureLoss == TYP.Chisholm then Modelica.Fluid.Dissipation.Utilities.Functions.PressureLoss.TwoPhase.dp_twoPhaseChisholm_DP(IN_con, IN_var, m_flow) else 0 "Frictional pressure loss";

`IN_con` has type `Modelica.Fluid.Dissipation.PressureLoss.StraightPipe.dp_twoPhaseOverall_IN_con` but the expected type was `Modelica.Fluid.Dissipation.Utilities.Records.General.TwoPhaseFlow_con` in call to function `Modelica.Fluid.Dissipation.Utilities.Functions.PressureLoss.TwoPhase.dp_twoPhaseFriedel_DP`.

According to [this](https://github.com/modelica/ModelicaStandardLibrary/issues/1245) calling a function with a record subtype of the original definition is invalid.

**Comment by HansOlsson** [comment]
TL:DR: Seems we have to use the old record constructor for this.

Full text:
Looking at the previous ticket it was originally about assignment and pointed to https://specification.modelica.org/v3.4/Ch10.html#equality-and-assignment and it isn't immediately clear that the same rules apply to function calls. However, https://specification.modelica.org/v3.4/Ch12.html#positional-or-named-input-arguments-of-functions currently specifies that.

Note that the new record constructor added in Modelica 3.4(?), https://specification.modelica.org/v3.4/Ch12.html#casting-to-record does not provide a simpler work-around as the argument cannot be a record. 

And changing the record type used in the function would break compatibility with existing calls outside of MSL.
